adaptive_scheduler/kernel/slicedipscheduler_v2.py

```diff
@@ -93,7 +93,13 @@ def build_data_structures(self):
                     scheduled = 1
                 # now w_idx is the index into r.possible_starts, which have
                 # been reordered by time.
-                self.Yik.append([r.resID, w_idx, r.priority, ps.resource, scheduled])
+                if r.request and r.request.optimization_type == 'AIRMASS':
```

Comment:
Non-functional comment, but defining the optimization types (and other things like this) as an enum is good way to avoid spelling mistakes and getting auto-complete suggestions from your editor.
```python
import enum

class OptimizationType(str, enum.Enum):
    AIRMASS = "AIRMASS"
    ... = "..."

...

if r.request.optimization_type == OptimizationType.AIRMASS:
    ...
```